### PR TITLE
:bug: Add annotation inspection support

### PR DESCRIPTION
**Pull request by jmle**
This PR adds support for inspecting annotations.

:green_circle: Things that can be done now:
- Check for annotations that are annotating specific `FIELD_DECLARATION`, `METHOD_DECLARATION` or `TYPE`
- Match against `METHOD_DECLARATION`

:red_circle: Things that cannot be done **yet**:
- Match `location: ANNOTATION` and inspect its values.
- Inspect on annotations on `location: CONSTRUCTOR`.

Depends on:
- https://github.com/konveyor/java-analyzer-bundle/pull/104

Fixes:
- https://github.com/konveyor/analyzer-lsp/issues/460
- https://github.com/konveyor/analyzer-lsp/issues/467

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func (r *ruleEngine) runTaggingRules(ctx context.Context, infoRules []ruleMessag
 				rs.Errors[rule.RuleID] = err.Error()
 			}
 		} else if response.Matched && len(response.Incidents) > 0 {
-			r.logger.V(5).Info("info rule was matched", "ruleID", rule.RuleID)
+			r.logger.V(5).Info("rule was matched", "ruleID", rule.RuleID)
```
Comment (shawn-hurley) [inline engine/engine.go:337]: why remove info here?

Comment (jmle) [reply engine/engine.go:337]: @shawn-hurley it sounded like a rule of type "info" was matched. It was confusing to me, as in windup we used to have INFO rules. But I can put it back if you want

File: external-providers/java-external-provider/pkg/java_external_provider/provider.go [hunk 2]
```diff
@@ -296,6 +308,7 @@ func (p *javaProvider) Init(ctx context.Context, log logr.Logger, config provide
 		"-Djava.net.useSystemProxies=true",
 		"-configuration",
 		"./",
+		//"--jvm-arg=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:1044",
```
Comment (jmle) [inline external-providers/java-external-provider/pkg/java_external_provider/provider.go:311]: Uncommenting this line allows for the LS to be run in debugging mode. I find it useful, but can be removed if you guys thing it's not convenient.

File: engine/engine.go (hunk 1 above)
Comment (shawn-hurley) [reply engine/engine.go:337]: can we updated to tagging rules, becuase they are seperate from the non-tagging rules

File: external-providers/java-external-provider/pkg/java_external_provider/provider.go (hunk 2 above)
Comment (shawn-hurley) [reply external-providers/java-external-provider/pkg/java_external_provider/provider.go:311]: nope that is useful! assume that allows you to attach a debugger and set break points and stuff?

File: engine/engine.go (hunk 1 above)
Comment (jmle) [reply engine/engine.go:337]: Then it probably makes sense to leave it as it was

**Review by pranavgaikwad — APPROVED** [review]

**Review by shawn-hurley — APPROVED** [review]

**Pull request merged**